extensions/amp-mowplayer/0.1/amp-mowplayer.js

```diff
@@ -0,0 +1,367 @@
+import {VideoEvents} from '../../../src/video-interface';
+import {addParamsToUrl} from '../../../src/url';
+import {
+  createFrameFor,
+  isJsonOrObj,
+  mutedOrUnmutedEvent,
+  objOrParseJson,
+  originMatches,
+  redispatch,
+} from '../../../src/iframe-video';
+import {dev, user} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {
+  fullscreenEnter,
+  fullscreenExit,
+  isFullscreenElement,
+  removeElement,
+} from '../../../src/dom';
+import {getData, listen} from '../../../src/event-helper';
+import {
+  installVideoManagerForDoc,
+} from '../../../src/service/video-manager-impl';
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+/**
+ * @enum {number}
+ * @private
+ */
+const PlayerStates = {
+  UNSTARTED: -1,
+  ENDED: 0,
+  PLAYING: 1,
+  PAUSED: 2,
+};
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpMowplayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?string}  */
+    this.mediaid_ = '';
+
+    /** @private {?boolean}  */
+    this.muted_ = false;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?string} */
+    this.videoIframeSrc_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    const {preconnect} = this;
+    preconnect.url(this.getVideoIframeSrc_());
+    // Host that mowplayer uses to serve JS needed by player.
+    preconnect.url('https://cdn.mowplayer.com', opt_onLayout);
+    // Load player settings
+    preconnect.url('https://code.mowplayer.com', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  viewportCallback(visible) {
+    this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible});
+  }
+
+  /** @override */
+  buildCallback() {
+    this.mediaid_ = user().assert(
+        (this.element.getAttribute('data-mediaid')),
+        '/The data-mediaid attribute is required for <amp-mowplayer> %s',
+        this.element);
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+  }
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+    const params = /** @type {!JsonObject} */ (dict({
+      'code': this.mediaid_,
+    }));
+    const src = addParamsToUrl('https://cdn.mowplayer.com/player.html',params);
```

Comment:
comma before `params`

Reply:
So basically need to remove typecast from line 129 and need to add typecast to line 132 like :